app/src/main/res/layout/dialog_persistent_location_permission.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical"
+    android:padding="16dp">
+
+    <TextView
+        android:id="@+id/tv_permission_title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Allow Location Always"
+        android:textSize="20sp"
+        android:textStyle="bold"
+        android:layout_marginBottom="16dp" />
+
+    <TextView
+        android:id="@+id/tv_permission_message"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="As in Android 12+ new functions have been introduced to facilitate Bluetooth Low Energy (BLE) scanning without requiring explicit location permissions. However, as our app is designed to support older Android versions, it remains necessary to request location permissions for BLE scanning on these systems. The scanning process used to detect nearby devices doesn't directly use GPS but requires access to location permissions on older Android versions for background scanning. This permission enables the app to scan for BLE packets while providing relative proximity data without revealing precise location information."
```

Comment:
The text already gives it away: we don’t need to request these permissions on Android 12+. We should only request that on old devices.